src/backend/main/py.main/caliopen_main/objects/message.py

```diff
@@ -116,6 +127,150 @@ def create_draft(cls, user_id=None, **params):
             raise exc
         return message
 
+    def patch_draft(self, patch, **options):
+        """operations specific to draft, before applying generic patch method"""
+
+        self.get_db()
+        self.unmarshall_db()
+        if not self.is_draft:
+            return err.PatchUnprocessable(message="this message is not a draft")
+
+        try:
+            params = dict(patch)
+            params.pop("current_state")
+            message_param = NewMessage(params)
+            message_param.validate()
+            valid_draft = Message.validate_draft_consistency(self.user_id,
+                                                             message_param,
+                                                             False)
+            print(valid_draft)
+        except Exception as exc:
+            log.warn(exc)
+            return err.PatchError(message=exc.message)
+
+        self.apply_patch(patch, **options)
+
+    @classmethod
+    def validate_draft_consistency(cls, user_id, draft, is_new):
```

Comment:
this method is too long and should be split in smaller methods for better readability.